src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs

```diff
@@ -288,6 +288,41 @@ internal TelemetryProcessorChain TelemetryProcessorChain
             }
         }
 
+        /// <summary>
+        /// This method will set the connection string, set the instrumentation key, validate the endpoints, and set the TelemetryChannel.Endpoint.
+        /// </summary>
+        /// <param name="connectionString">Users unique connection string. This must include the Instrumentation Key. Format: "key1=value1;key2=value2;key3=value3".</param>
+        /// <exception cref="ConnectionStringDuplicateKeyException">Thrown when the connection string has duplicate keys.</exception>
+        /// <exception cref="ConnectionStringInvalidDelimiterException">Thrown when the connection string is formatted incorrectly.</exception>
+        /// <exception cref="ConnectionStringInvalidEndpointException">Thrown when the endpoint that was provided is invalid. Exception contains more information.</exception>
+        /// <exception cref="ConnectionStringMissingInstrumentationKeyException">Thrown when the connection string does not contain an instrumentation key (required).</exception>
+        public void SetConnectionString(string connectionString)
+        {
+            try
+            {
+                this.ConnectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
+
+                var endpointProvider = new EndpointProvider
+                {
+                    ConnectionString = connectionString,
+                };
+
+                this.InstrumentationKey = endpointProvider.GetInstrumentationKey();
+
+                this.Endpoint = new EndpointContainer(endpointProvider);
+
+                foreach (var tSink in this.TelemetrySinks)
+                {
+                    tSink.TelemetryChannel.EndpointAddress = new Uri(this.Endpoint.Ingestion, "v2/track").AbsoluteUri;
```

Comment:
why all sinks should get the same endpoint? My guess is sinks ARE intended to send data to different places, e.g.: one to AppInsights, another to Geneva. (check out [this](https://1dsdocs.azurewebsites.net/articles/concepts/telemetry-sink.html) or [this](https://1dsdocs.azurewebsites.net/getting-started/csharp-getting_started.html#geneva-channel) )

 I'd assume only default one should have AI endpoint at all or only our channels - InMem and Server should get it.